### PR TITLE
[REF] website_sale, website_portal: Re-use partner.check_vat method

### PR DESCRIPTION
**Pull request by moylop260**
Avoid copy and paste code and repeated lines if a change is generated
from check_vat so we need to remember upgrade a copy and paste of all
models where is used this validation.

Use check_vat function directly help us to re-use 100% the code without
errors and inheritable.

"A good developer create code a better one re-use":
re-use

### CODE REVIEW

**Comment by moylop260** [comment]
FYI @qdp-odoo this PR
 - Allow inherit for`base_vat` module
 - Create consistence with last changes to validate VAT based on country (and other futures native changes)
 - Avoid differences for backend and fronted logic to validate VAT
 - Re-use code
 - Remove ugly copy&paste practice

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -485,13 +485,15 @@ def checkout_form_validate(self, mode, all_form_values, data):
         # vat validation
         Partner = request.env['res.partner']
         if data.get("vat") and hasattr(Partner, "check_vat"):
-            check_func = request.website.company_id.vat_check_vies and Partner.vies_vat_check or Partner.simple_vat_check
-            vat_country, vat_number = Partner._split_vat(data.get("vat"))
-            if not check_func(vat_country, vat_number):
-                if country.code and check_func(country.code, data["vat"]):
```
Comment (mart-e) [inline addons/website_sale/controllers/main.py]: you are changing the behaviour here. at bef9b1114c2e8a15e8a87e1eb5f0451f93ec75d3 we allowed to use both format with and without the country code

Comment (moylop260) [inline addons/website_sale/controllers/main.py]: Thanks for reply @mart-e 
Sorry, I didn't understand what is the difference of the behaviour between this PR and https://github.com/odoo/odoo/commit/bef9b1114c2e8a15e8a87e1eb5f0451f93ec75d3

I  mean, the original [partner.check_vat](https://github.com/odoo/odoo/blob/2dcf233a2801a6fda0cb03f47aca83182b7803ce/addons/base_vat/models/res_partner.py#L100) supports both format with and without the country code (thanks @qdp-odoo).

Could you help me to understand it, please?

Comment (mart-e) [inline addons/website_sale/controllers/main.py]: Oh right, I did not know the check_vat did that (it was done recently). So yes it makes sense to harmonize.

**Comment by mart-e** [comment]
Thanks, merged at https://github.com/odoo/odoo/commit/d7f84c273b6584606451d09bc4f0f7e091c4096c

**Pull request merged**